### DISCUSSION TITLE
Custom topbar (navbar) component to specific panel

### DISCUSSION THREAD

**Discussion by shoxrux1996** [filamentphp/filament#8889] [Help] [answered]
I want to change topbar of the panel with my new custom component. If I add this new component to the folder with path: `resources/views/vendor/filament-panels/components/topbar/`, it works. However it updates all panels. 

Lets say I have **customer** panel, and I want to update navbar only **customer panel**. Is there any way to configure the panel, so that it would only take components from this path: `resources/views/filament/cusomer/vendor/filament-panels/**/*.blade.php`?

**Comment by leandrocfe** [comment, marked as answer]
maybe checking the panel id?

```html
<div>
    @if (filament()->getCurrentPanel()->getId() === 'admin')
        {{-- include a --}}
    @else
        {{-- include b --}}
    @endif
</div>
```

**Reply by shoxrux1996** [reply]
Thank you, it worked. I copied filament-panels layout and updated to that :
```
            @if (filament()->getCurrentPanel()->getId() === 'customer')
            <x-filament-panels::topbar-custom :navigation="$navigation" />
            @else
            <x-filament-panels::topbar :navigation="$navigation" />
            @endif
```
 Now I can include custom component using panel id.  I know it's bad practice to update the resources/views/vendor/ because of frequent updates. But I need  to have the header fully customizable

**Comment by khalidmaquilang** [comment]
@shoxrux1996 how did you do it? i want to modify my top navbar also

**Reply by soofdev** [reply]
Same here... could you explain the entire approach? Thanks.

**Reply by khalidmaquilang** [reply]
actually I just added a hook on the provider..

```
->renderHook(
                PanelsRenderHook::GLOBAL_SEARCH_BEFORE,
                fn (): string => Blade::render('@livewire(\'member.cart\')'),
            )
            ->renderHook(
                PanelsRenderHook::CONTENT_START,
                fn (): string => Blade::render('@livewire(\'member.cart-sidebar\')'),
            )
            ->renderHook(
                PanelsRenderHook::PAGE_HEADER_ACTIONS_BEFORE,
                fn (): string => Blade::render('@livewire(\'member.invitation-link-button\')'),
                scopes: Dashboard::class,
            );
```

global search is on the topbar

**Reply by khalidmaquilang** [reply]
my goal was to add a cart.. not really a full custom topbar

**Comment by shoxrux1996** [comment]
Filament (panel package) comes with its own set of views, just like other Laravel packages. Instead of directly modifying the package views in vendor/, I copied the necessary views to my own resources folder where they can be safely modified.

Here's what I did:
First, I copied the topbar view from vendor/filament/filament/resources/views/components/topbar.blade.php to my own resources/views/vendor/filament-panels/components/topbar.blade.php.
Similarly, I copied the layout file from vendor/filament/filament/resources/views/components/layout.blade.php into resources/views/vendor/filament-panels/components/layout.blade.php and made my changes there.

You can also publish all the Filament views by running:

`php artisan vendor:publish --tag=filament-panels-views`

This will copy all the package views to your project, allowing you to modify them as needed.
Important Note: It’s not recommended to edit core package views directly in the vendor/ directory because they may be overwritten during package updates. Always make changes to the published views in your resources/views folder to ensure they persist across upgrades.
If you want to explore the default views, you can check them in the vendor/filament/filament/resources/views folder.
